### DISCUSSION TITLE
cookieconsent and vite-ssg

### DISCUSSION THREAD

**Discussion by sebbm** [orestbida/cookieconsent#437] [Q&A] [answered]
Hi all,
maybe somebody can help me to use cookiesonsent and vite-ssg. If i integrate cookieconsent like described in the documentation: 
[https://cookieconsent.orestbida.com/essential/getting-started.html#vue](https://cookieconsent.orestbida.com/essential/getting-started.html#vue) it runs in client operation mode. When i try to build it for ssg then building process is aborted with error message: 

app.config.globalProperties.$cc.run(pluginConfig);
                                                   ^
TypeError: app.config.globalProperties.$cc.run is not a function

Any idea how to get this working ?

**Comment by orestbida** [comment, marked as answer]
Hi @sebbm,

create a `cookieconsent.vue` component — where you setup the plugin — and then wrap it with 
`<client-only>` tags when you import it in the main component.

I added a [demo example on stackblitz](https://stackblitz.com/edit/vitejs-vite-erhdeg?file=src%2FApp.vue).

**Reply by sebbm** [reply]
HI @orestbida , thank you so much ! This example really helped me to understand the logic !

**Comment by sebbm** [comment]
@orestbida , May i ask you for an additional hint ? How can i access the instance in composition api from a seperate route ? My intention is to hide the cookiebar for the data privacy page and impressum but i don't know how to access the CookieConsent from the route components
